Refactor task names

BenjaminLudwigSAP opened the pull request:
Refactor some task names to make more immediately clear what they do.

Please squash-merge.

BenjaminLudwigSAP committed:
[l2] Refactor: Rename guest VLAN (un)assignment tasks

- Rename EnsureGuestVLAN to AssignVLANToGuest
- Rename RemoveGuestVLAN to UnassignVLANFromGuest

BenjaminLudwigSAP committed:
[l2] Refactor: Rename VLAN removal task

- Rename RemoveVLANIfNotOwnedByGuest to RemoveVLANIfNotOwnedByOtherGuest

To make clear that it's other guests who might still be using this VLAN.

BenjaminLudwigSAP committed:
[l2] Refactor: Improve comment about not unassigning VLAN from LAG

It sounded like a suggestion, but it's supposed to be a warning.

m-kratochvil commented on octavia_f5/controller/worker/flows/f5_flows_iseries.py:
```diff
@@ -302,24 +302,25 @@ def make_ensure_vcmp_l2_flow(self) -> flow.Flow:
         ensure_vlan = self.tasks.EnsureVLAN()
         ensure_vlan_interface = self.tasks.EnsureVLANInterface()
-        ensure_guest_vlan = self.tasks.EnsureGuestVLAN()
+        assign_vlan_to_guest = self.tasks.AssignVLANToGuest()
```
We could also consider keeping the `ensure...` naming convention consistent and use `ensure_vlan_to_guest_assignment`. What do you think?

BenjaminLudwigSAP replied:
I renamed both the assignment and unassignment tasks to stick to the Ensure/Remove naming scheme.

m-kratochvil commented on octavia_f5/controller/worker/flows/f5_flows_iseries.py:
```diff
@@ -302,24 +302,25 @@ def make_ensure_vcmp_l2_flow(self) -> flow.Flow:
-        # no need to unassign the VLAN from the interface/LAG, because that is
-        # going to happen at VLAN deletioni anyway.
-        remove_vlan_if_not_owned_by_guest = self.tasks.RemoveVLANIfNotOwnedByGuest()
+        remove_vlan_from_guest = self.tasks.UnassignVLANFromGuest()
```
The `remove_vlan_from_guest` is a bit confusing here. It could be interpreted as "removing (deleting) a vlan from the actual guest device. I know that this is here in context of the vcmp host flow but still. I would suggest renaming this to `unassign_vlan_from_guest`, it would then also match the task name.

BenjaminLudwigSAP replied:
Oh. Apparently I simply forgot to rename that variable. Of course I want to have them match the task names.

BenjaminLudwigSAP committed:
[l2] Refactor VLAN guest assignment tasks: Keep Ensure/Remove scheme

- Rename AssignVLANToGuest to EnsureVLANGuestAssignment

- Rename UnassignVLANFromGuest to RemoveVLANGuestAssignment

Also: Fix variable name to match the name of the task class it's an
instance of:
- Rename assign_vlan_to_guest to ensure_vlan_guest_assignment
- Rename remove_vlan_from_guest to remove_vlan_guest_assignment

BenjaminLudwigSAP committed:
[l2] Refactor VLAN guest assignment tasks: Make error messages clearer

Same reason as why the tasks were renamed.

m-kratochvil reviewed (approved):
Looks great!
